<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataGenerator</class>
 <widget class="QWidget" name="DataGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>543</width>
    <height>349</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Generator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="2" column="1">
    <widget class="QLineEdit" name="le_outFile"/>
   </item>
   <item row="1" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0" colspan="2">
       <widget class="QStackedWidget" name="sw_mode">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="pg_rawData">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Amount:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spn_rawData_amount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>999</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="cmb_rawData_units">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="currentText">
             <string>Megabytes</string>
            </property>
            <item>
             <property name="text">
              <string>Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Kilobytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Megabytes</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pg_distribution">
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cmb_distribution">
            <item>
             <property name="text">
              <string>Uniform</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal (Gaussian)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Geometric</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Exponential</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Poisson</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lbl_distributionType">
            <property name="text">
             <string>Distribution:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cmb_distributionType">
            <item>
             <property name="text">
              <string>Continuous</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Discrete</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>N:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spn_n">
            <property name="toolTip">
             <string>The number of data points to generate</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000000000</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QStackedWidget" name="sw_params">
            <widget class="QWidget" name="pg_uniform">
             <layout class="QFormLayout" name="formLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Minimum:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Maximum:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="spn_u_min">
                <property name="toolTip">
                 <string>The uniform range minimum</string>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="minimum">
                 <double>-1000000.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="spn_u_max">
                <property name="toolTip">
                 <string>The uniform range maximum</string>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="minimum">
                 <double>-1000000.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
                <property name="value">
                 <double>1.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="pg_normal">
             <layout class="QFormLayout" name="formLayout_6">
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Mean:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Ïƒ:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="spn_normal_sigma">
                <property name="toolTip">
                 <string>The normal distribution standard deviation</string>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
                <property name="value">
                 <double>10.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="spn_normal_mean">
                <property name="toolTip">
                 <string>The normal distribution mean</string>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="minimum">
                 <double>-1000000.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="pg_geometric">
             <layout class="QFormLayout" name="formLayout_7">
              <item row="0" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Expected Value:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="spn_geometric_e">
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
                <property name="value">
                 <double>10.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="pg_exponential">
             <layout class="QFormLayout" name="formLayout_8">
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Lambda:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="spn_exponential_lambda">
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
                <property name="value">
                 <double>10.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="pg_poisson">
             <layout class="QFormLayout" name="formLayout_9">
              <item row="0" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Expected Value:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="spn_poisson_e">
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="maximum">
                 <double>1000000.000000000000000</double>
                </property>
                <property name="value">
                 <double>10.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QComboBox" name="cmb_mode">
     <item>
      <property name="text">
       <string>Distribution</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Raw Data</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QToolButton" name="toolButton">
     <property name="text">
      <string>...</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>Output File:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Mode:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="3">
    <widget class="QPushButton" name="btn_generate">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Generate</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btn_generate</sender>
   <signal>clicked()</signal>
   <receiver>DataGenerator</receiver>
   <slot>_generate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmb_mode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>DataGenerator</receiver>
   <slot>_mode_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmb_distribution</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>DataGenerator</receiver>
   <slot>_distribution_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmb_distributionType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>DataGenerator</receiver>
   <slot>_distribution_type_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>DataGenerator</receiver>
   <slot>_select_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>289</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_generate()</slot>
  <slot>_mode_changed(int)</slot>
  <slot>_distribution_changed(int)</slot>
  <slot>_distribution_type_changed(int)</slot>
  <slot>_select_file()</slot>
 </slots>
</ui>
